pkg/splunk/enterprise/util.go

```diff
@@ -211,6 +217,85 @@ func GetSmartstoreRemoteVolumeSecrets(volume enterprisev1.VolumeSpec, client spl
 	return accessKey, secretKey, namespaceScopedSecret.ResourceVersion, nil
 }
 
+// ApplyAppListingConfigMap creates the configMap  with two entries:
+// (1) app-list-local.yaml
+// (2) app-list-cluster.yaml
+func ApplyAppListingConfigMap(client splcommon.ControllerClient, cr splcommon.MetaObject,
+	appConf *enterprisev1.AppFrameworkSpec, appsSrcDeployStatus map[string]enterprisev1.AppSrcDeployInfo) (*corev1.ConfigMap, bool, error) {
+
+	var err error
+	var crKind string
+	var configMapDataChanged bool
+	crKind = cr.GetObjectKind().GroupVersionKind().Kind
+
+	scopedLog := log.WithName("ApplyAppListingConfigMap").WithValues("kind", crKind, "name", cr.GetName(), "namespace", cr.GetNamespace())
+
+	mapAppListing := make(map[string]string)
+
+	// ToDo: sgontla/Jeff: Adapt to any Ansible changes(if required) for differentiating local vs. cluster scope (?)
+	yamlConfHeader := fmt.Sprintf(`splunk:
+  apps_location:`)
+
+	var localAppsConf, clusterAppsConf string
+	localAppsConf = yamlConfHeader
+	clusterAppsConf = yamlConfHeader
+
+	for appSrc, appSrcDeployInfo := range appsSrcDeployStatus {
+		appDeployList := appSrcDeployInfo.AppDeploymentInfoList
+
+		switch scope := getAppSrcScope(appConf, appSrc); scope {
+		case "local":
+			for idx := range appDeployList {
+				if appDeployList[idx].DeployStatus == enterprisev1.DeployStatusPending {
+					localAppsConf = fmt.Sprintf(`%s
+    - "/init-apps/%s/%s"`, localAppsConf, appSrc, appDeployList[idx].AppName)
+				}
+			}
+
+		case "cluster":
+			for idx := range appDeployList {
+				if appDeployList[idx].DeployStatus == enterprisev1.DeployStatusPending {
+					clusterAppsConf = fmt.Sprintf(`%s
+    - "/init-apps/%s/%s"`, clusterAppsConf, appSrc, appDeployList[idx].AppName)
+				}
+			}
+
+		default:
+			scopedLog.Error(nil, "Invalid scope detected")
+		}
+	}
+
+	if localAppsConf != yamlConfHeader {
+		mapAppListing["app-list-local.yaml"] = localAppsConf
+	}
+
+	if clusterAppsConf != yamlConfHeader {
+		mapAppListing["app-list-cluster.yaml"] = clusterAppsConf
+	}
+
+	// Create App list config map
+	configMapName := GetSplunkAppsConfigMapName(cr.GetName(), crKind)
+	SplunkAppListingConfigMap := prepareConfigMap(configMapName, cr.GetNamespace(), mapAppListing)
+
+	SplunkAppListingConfigMap.SetOwnerReferences(append(SplunkAppListingConfigMap.GetOwnerReferences(), splcommon.AsOwner(cr, true)))
+	configMapDataChanged, err = splctrl.ApplyConfigMap(client, SplunkAppListingConfigMap)
+	if err != nil {
+		return nil, configMapDataChanged, err
+	} else if configMapDataChanged {
+		// Create a token to check if the app list is on the pod is latest or not
+		mapAppListing[appsUpdateToken] = fmt.Sprintf(`%d`, time.Now().Unix())
+
+		// Apply the configMap with a fresh token, if there is a change
+		SplunkAppListingConfigMap = prepareConfigMap(configMapName, cr.GetNamespace(), mapAppListing)
```

Comment:
Correct me please, so on data change in configmap we are preparing a new configmap? and then Applying it? 
We can just use ApplyConfigMap() to pick new changes right?

Reply:
Good catch. Ya, that should work. we don't need to call the prepare twice. Will remove it.